### ISSUE TITLE
Report unresolved imports in the editor

### ISSUE THREAD

**Issue by casper-bach** [metaborg/dynsem#56]
The following program seems to crash something in the analysis pipeline:

```
module test/coq-semantics/features/import1

imports importx
```

Crashes when trying any transformation menu-item in `Spoofax > Semantics`

Removing the import fixes the error. Any idea why?

Error:

```
org.metaborg.core.transform.TransformException: Invoking Stratego strategy module-to-core-editor failed at term
    ( Module("test/coq-semantics/features/import1", [Imports([Import("importx")])])
, []
, Module("test/coq-semantics/features/import1", [Imports([Import("importx")])])
, "test/coq-semantics/features/import1.ds"
, "eclipse:///dynsem"
)
Stratego trace:
    module_to_core_editor_0_0
    module_to_core_editor_0_0
    report_with_failure_0_1
    report_failure_0_2
    SRTS_EXT_fatal_err_0_2

    at org.metaborg.spoofax.core.transform.StrategoTransformer.transform(StrategoTransformer.java:136) ~[org.metaborg.spoofax.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.core.transform.StrategoTransformer.transform(StrategoTransformer.java:63) ~[org.metaborg.spoofax.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.core.transform.StrategoTransformer.transform(StrategoTransformer.java:1) ~[org.metaborg.spoofax.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.core.transform.TransformService.transform(TransformService.java:51) ~[org.metaborg.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.eclipse.transform.TransformJob.transform(TransformJob.java:141) [org.metaborg.spoofax.eclipse_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.eclipse.transform.TransformJob.transformAll(TransformJob.java:108) [org.metaborg.spoofax.eclipse_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.eclipse.transform.TransformJob.run(TransformJob.java:70) [org.metaborg.spoofax.eclipse_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54) [org.eclipse.core.jobs_3.6.1.v20141014-1248.jar:na]
Caused by: org.spoofax.interpreter.core.InterpreterErrorExit: Internal error: with clause failed unexpectedly in 'module-to-core-editor'
    (Module("test/coq-semantics/features/import1",[Imports([Import("importx")])]),[],Module("test/coq-semantics/features/import1",[Imports([Import("importx")])]),"test/coq-semantics/features/import1.ds","eclipse:///dynsem")
    at org.strategoxt.lang.InteropSDefT.evaluate(InteropSDefT.java:194) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.strategoxt.lang.InteropSDefT.evaluate(InteropSDefT.java:183) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.strategoxt.lang.InteropSDefT$StrategyBody.evaluate(InteropSDefT.java:245) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.spoofax.interpreter.core.Interpreter.evaluate(Interpreter.java:109) ~[org.spoofax.interpreter.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.spoofax.interpreter.core.Interpreter.invoke(Interpreter.java:82) ~[org.spoofax.interpreter.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.strategoxt.HybridInterpreter.invoke(HybridInterpreter.java:442) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.metaborg.spoofax.core.stratego.StrategoCommon.invoke(StrategoCommon.java:78) ~[org.metaborg.spoofax.core_2.0.0.20160527-092423-DETACHED.jar:na]
    at org.metaborg.spoofax.core.transform.StrategoTransformer.transform(StrategoTransformer.java:134) ~[org.metaborg.spoofax.core_2.0.0.20160527-092423-DETACHED.jar:na]
    ... 7 common frames omitted
Caused by: org.strategoxt.lang.StrategoErrorExit: Internal error: with clause failed unexpectedly in 'module-to-core-editor'
    at org.strategoxt.lang.SRTS_EXT_fatal_err_0_3.invoke(SRTS_EXT_fatal_err_0_3.java:28) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.strategoxt.lang.compat.override.java_integration.SRTS_EXT_fatal_err_0_2.invoke(SRTS_EXT_fatal_err_0_2.java:33) ~[strategoxt-min-jar-2.0.0-SNAPSHOT.jar:na]
    at org.strategoxt.lang.compat.override.java_integration.report_failure_0_2_override.invoke(report_failure_0_2_override.java:27) ~[strategoxt-min-jar-2.0.0-SNAPSHOT.jar:na]
    at org.strategoxt.stratego_lib.report_with_failure_0_1.invoke(report_with_failure_0_1.java:32) ~[strategoxt-min-jar-2.0.0-SNAPSHOT.jar:na]
    at trans.module_to_core_editor_0_0.invoke(module_to_core_editor_0_0.java:69) ~[na:na]
    at org.strategoxt.lang.Strategy.invokeDynamic(Strategy.java:30) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    at org.strategoxt.lang.InteropSDefT.evaluate(InteropSDefT.java:192) ~[org.strategoxt.strj_2.0.0.20160527-092423-DETACHED/:na]
    ... 14 common frames omitted
```

**Comment by vvergu** [comment]
Hi @casperbp. I think the import cannot be resolved. Normally we import with a qualified path relative to the project root, in your case i think the import would look like `import test/coq-semantics/features/importx`. Can you try to see if this resolves the crash?

If that import cannot be resolved and if using a qualified path does fix, the analysis shouldn't crash to badly. A message should be displayed in the editor in the first place.

**Comment by casper-bach** [comment]
Thanks! Using the fully qualified path fixes it. Hadn't realised it needs fully qualified paths. An error message about missing import(s) would be good.

**Comment by vvergu** [comment]
Thanks for the confirmation. I'll mark the issue appropriately.

**Comment by vvergu** [comment]
Thank you @casperbp for reporting this issue. I've fixed this in c7b7524. The commit also includes other improvements to import failure handling.
